packages/frontend/src/features/semanticViewer/api/hooks.ts

```diff
@@ -26,15 +26,31 @@ export const useSemanticLayerViews = ({
 type SemanticLayerViewFieldsParams = {
     projectUuid: string;
     view: string;
+    selectedFields: Pick<
+        SemanticLayerQuery,
+        'dimensions' | 'timeDimensions' | 'metrics'
+    >;
 };
 
 export const useSemanticLayerViewFields = ({
     projectUuid,
     view,
+    selectedFields,
 }: SemanticLayerViewFieldsParams) =>
     useQuery<SemanticLayerField[], ApiError>({
-        queryKey: [projectUuid, 'semanticLayer', view, 'fields'],
-        queryFn: () => apiGetSemanticLayerViewFields({ projectUuid, view }),
+        queryKey: [
+            projectUuid,
+            'semanticLayer',
+            view,
+            'fields',
+            selectedFields,
```

Comment:
do we need to serialize this?

Reply:
I believe `useQuery` already does that for us: `When a query needs more information to uniquely describe its data, you can use an array with a string and any number of serializable objects to describe it`: https://tanstack.com/query/v4/docs/framework/react/guides/query-keys